client/src/main/java/org/apache/uniffle/client/impl/ShuffleWriteClientImpl.java

```diff
@@ -520,18 +520,26 @@ public void reportShuffleResult(
     Map<ShuffleServerInfo, List<Integer>> groupedPartitions = Maps.newHashMap();
     Map<Integer, Integer> partitionReportTracker = Maps.newHashMap();
     for (Map.Entry<Integer, List<ShuffleServerInfo>> entry : partitionToServers.entrySet()) {
+      int partitionIdx = entry.getKey();
       for (ShuffleServerInfo ssi : entry.getValue()) {
-        if (!groupedPartitions.containsKey(ssi)) {
-          groupedPartitions.putIfAbsent(ssi, Lists.newArrayList());
-        }
-        groupedPartitions.get(ssi).add(entry.getKey());
+        groupedPartitions.putIfAbsent(ssi, Lists.newArrayList());
```

Comment:
`Lists.newArrayList()` will be invoked everytime.

Reply:
Emm.. Yes, I forget this problem for `putIfAbsent`

Reply:
Actually, the original logic is OK for me. Let me recover this part.